applications/StructuralMechanicsApplication/custom_elements/prestress_membrane_element.cpp

```diff
@@ -1132,142 +1135,552 @@ void PrestressMembraneElement::CalculateMembraneElasticityTensor(
+    mDetJ0.resize(integration_points.size());
+    mGab0.resize(integration_points.size());
+    mG1.resize(integration_points.size());
+    mG2.resize(integration_points.size());
+
+    GeometryType::JacobiansType J0;
+    J0 = GetGeometry().Jacobian(J0);
+
+    //KRATOS_WATCH(J0);
 
-    // Then T2 can only be the cross product between T1 and T3
-    CrossProduct(t2, t3, t1);
+    mStrainsVector.resize(integration_points.size());
+    mStressesVector.resize(integration_points.size());
+    mCauchyStressesVector.resize(integration_points.size()); //VM
 
-    //if (this->GetId() == 42 && PointNumber == 1)
-    //    KRATOS_WATCH(t2);
+    mG_Vector.resize(integration_points.size(), ZeroMatrix(2, 2));
 
+    // initialize prestress matrix and prestress directions
+    Matrix& prestress_variable = this->GetValue(MEMBRANE_PRESTRESS);
+    Matrix& prestress_axis_1 = this->GetValue(PRESTRESS_AXIS_1_GLOBAL);
+    Matrix& prestress_axis_2 = this->GetValue(PRESTRESS_AXIS_2_GLOBAL);
+    unsigned int strain_size = this->GetProperties().GetValue(CONSTITUTIVE_LAW)->GetStrainSize();
 
-    // Normalization of the vectors
-    t1 = t1 / norm_2(t1);
-    t2 = t2 / norm_2(t2);
-    t3 = t3 / norm_2(t3);
 
-    // Transformation matrix from the projected basis T to the local cartesian basis
-    double eG11 = inner_prod(e1, t1);
-    double eG12 = inner_prod(e1, t2);
-    double eG21 = inner_prod(e2, t1);
-    double eG22 = inner_prod(e2, t2);
+    // Calculating geometry tensors in reference configuration on Integration points
+    for (unsigned int PointNumber = 0; PointNumber < integration_points.size(); PointNumber++)
+    {
+        // update prestress (in case of anisotropic prestress)
+        // if step=1: no update necessary
+        if(mAnisotropicPrestress == true && rCurrentProcessInfo[TIME] != 1){
+            UpdatePrestress(PointNumber);
+        }
 
-    //if (this->GetId() == 42 && PointNumber == 1)
-    //{
-    //    KRATOS_WATCH(eG11);
-    //    KRATOS_WATCH(eG12);
-    //    KRATOS_WATCH(eG21);
-    //    KRATOS_WATCH(eG22);
-    //}
 
+        // getting information for integration
+        double IntegrationWeight = integration_points[PointNumber].Weight();
+        //std::cout << "PrestressMembraneElement::Initialize(): 2 " << this->GetId() << std::endl;
+        // base vectors
+        array_1d<double, 3> g1;
+        array_1d<double, 3> g2;
+        array_1d<double, 3> g3;
+
+        g1[0] = J0[PointNumber](0, 0);
+        g2[0] = J0[PointNumber](0, 1);
+        g1[1] = J0[PointNumber](1, 0);
+        g2[1] = J0[PointNumber](1, 1);
+        g1[2] = J0[PointNumber](2, 0);
+        g2[2] = J0[PointNumber](2, 1);
+
+
+        // Store base vectors in reference configuration
+        mG1[PointNumber] = g1;
+        mG2[PointNumber] = g2;
+        
+
+        //KRATOS_WATCH(J0);
+
+        // base vector g3
+        CrossProduct(g3, g1, g2);
+        // differential area dA
+        const double dA = norm_2(g3);
+
+        if (dA == 0)
```

Comment:
Do not compare double with `==`, this will fail
Compare it with `<` or `>`

In your case I would use:
`KRATOS_ERROR_IF(dA < std::numeric_limits<double>::epsilon()) << "PrestressMembraneElement with id" << this->GetId() << "has ZERO differential area!" << std::endl;`